### ISSUE TITLE
Transfer assignment

### ISSUE THREAD

**Issue by maanlamp** [cmda-be/course-17-18#480]
# HTTP Cheatsheet

### OPTIONS
The options method tells the server to respond with all existing methods applicable to te provided URL.

### HEAD
The head method returns all headers you would get if you were to send a GET request, but it also specifically tells the server to not send any data back. This is useful to check if a resource exists, or to get specific header data without having the server return an entire response body.

### GET
The get method simply gets data from the provided URL.

### POST
The post method tells the server to store data sent with the request as a subordinate of the given URL.

### PUT
The put method beaves a bit like post, but put should only put something in a resource, or overwrite it. It does not insert it as a subordinate resource like post does.

### PATCH
The patch method is used to patch a piece of data in a given resource.

### DELETE
The delete method lets you freely delete resources, if the server doesn't have a security protocol.

## Status codes
I didn't specifically learn new status codes, since [those can easily be found everywhere](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes).

## HTTP headers
I learnt to use certain headers such as 'Accept', 'Accept-encoding', 'Content-type' and 'Authorization'.

## Curl
Curl was easy to use except in case of some requests, where I ad to look up how to do certain things. The documentation for curl specifies different syntax order in some cases, which was only a bit confusing at most.

In conclusion: I think it went `200: OK` 👌

**Comment by wooorm** [comment]
Nicely done!
